src/cartridges/int_adyen_controllers/cartridge/controllers/Adyen.js

```diff
@@ -426,17 +424,28 @@ function showConfirmationPaymentFromComponent() {
     adyenPaymentInstrument.custom.adyenPaymentData = null;
   });
 
-  finalResult = finalResult || adyenCheckout.doPaymentsDetailsCall(requestObject);
-
-  if (['Authorised', 'Pending', 'Received'].indexOf(finalResult.resultCode) > -1) {
+  if (order.status.value === Order.ORDER_STATUS_CREATED) {
+    finalResult = finalResult || adyenCheckout.doPaymentsDetailsCall(requestObject);
```

Comment:
Can we introduce an additional value to get rid of `finalResult = finalResult`? I would like to suggest introducing `amazonPayResult` at line 389 and to introduce finalResult at line 426.
I think that would make this class a little less confusing

Reply:
Will do so!